### PR TITLE
#168 Update docker modules to use fabric8

### PR DESCRIPTION
**Pull request by carter-cundiff**
#168

### CODE REVIEW

File: build-parent/pom.xml [hunk 1]
```diff
@@ -74,6 +74,10 @@
         <version.quarkus.cucumber>0.6.0</version.quarkus.cucumber>
         <version.quarkus.cucumber.java>7.4.1</version.quarkus.cucumber.java>
 
+        <!-- io.fabric8:docker-maven-plugin -->
+        <version.fabric8.docker.maven.plugin>0.44.0</version.fabric8.docker.maven.plugin>
+        <docker.platforms>linux/amd64,linux/arm64</docker.platforms>
```
Comment (carter-cundiff) [inline build-parent/pom.xml:79]: I: Fabric8 will only build your local machines architecture when building on `mvn clean install` so safe to list both platforms here.

File: extensions/extensions-docker/aissemble-model-training-api-containers/aissemble-model-training-api-sagemaker/pom.xml [hunk 2]
```diff
@@ -36,14 +32,6 @@
...
                         <configuration>
                             <wheelDependencies>
-                                <wheelDependency>
-                                    <artifactId>aissemble-foundation-core-python</artifactId>
-                                    <targetDirectory>${wheelOutputDirectory}</targetDirectory>
-                                </wheelDependency>
-                                <wheelDependency>
-                                    <artifactId>aissemble-foundation-pdp-client-python</artifactId>
-                                    <targetDirectory>${wheelOutputDirectory}</targetDirectory>
-                                </wheelDependency>
```
Comment (carter-cundiff) [inline extensions/extensions-docker/aissemble-model-training-api-containers/aissemble-model-training-api-sagemaker/pom.xml:46]: I: Removed unneeded dependencies

File: extensions/extensions-docker/aissemble-model-training-api-containers/aissemble-model-training-api-sagemaker/pom.xml [hunk 3]
```diff
@@ -85,25 +73,12 @@
...
-            <artifactId>aissemble-foundation-core-python</artifactId>
-            <version>${project.version}</version>
-            <type>pom</type>
-        </dependency>
-        <dependency>
-            <groupId>com.boozallen.aissemble</groupId>
-            <artifactId>aissemble-foundation-pdp-client-python</artifactId>
-            <version>${project.version}</version>
-            <type>pom</type>
-        </dependency>
```
Comment (carter-cundiff) [inline extensions/extensions-docker/aissemble-model-training-api-containers/aissemble-model-training-api-sagemaker/pom.xml:99]: I: Same as above

File: extensions/extensions-docker/aissemble-model-training-api-containers/aissemble-model-training-api/pom.xml [hunk 4]
```diff
@@ -89,29 +77,11 @@
...
-            <artifactId>aissemble-foundation-pdp-client-python</artifactId>
-            <version>${project.version}</version>
-            <type>pom</type>
-        </dependency>
-        <dependency>
-            <groupId>com.boozallen.aissemble</groupId>
-            <artifactId>aissemble-foundation-messaging-python-client</artifactId>
-            <version>${project.version}</version>
-            <type>pom</type>
-        </dependency>
```
Comment (carter-cundiff) [inline extensions/extensions-docker/aissemble-model-training-api-containers/aissemble-model-training-api/pom.xml:109]: I: Same as above

File: extensions/extensions-docker/aissemble-versioning/pom.xml [hunk 5]
```diff
@@ -79,23 +79,11 @@
...
-            <artifactId>aissemble-foundation-core-python</artifactId>
-            <version>${project.version}</version>
-            <type>pom</type>
-        </dependency>
-        <dependency>
-            <groupId>com.boozallen.aissemble</groupId>
-            <artifactId>aissemble-foundation-pdp-client-python</artifactId>
-            <version>${project.version}</version>
-            <type>pom</type>
-        </dependency>
```
Comment (carter-cundiff) [inline extensions/extensions-docker/aissemble-versioning/pom.xml:93]: I: Removed unnecessary dependencies

File: extensions/extensions-docker/pom.xml [hunk 6]
```diff
@@ -99,20 +106,24 @@
...
-                <configuration>
-                    <buildArgs>
-                        <!-- allows arch-specific builds to reference the exact base image, might not be needed once we do multi-arch builds + manifest in Orphedomos -->
-                        <VERSION_AISSEMBLE>${dockerImageVersion}</VERSION_AISSEMBLE>
-                        <DELTA_HIVE_CONNECTOR_VERSION>${version.delta.hive.connector}</DELTA_HIVE_CONNECTOR_VERSION>
-                    </buildArgs>
-                    <aliases>
-                        <!-- Allows downstream projects/modules to use the locally built image -->
-                        <!-- It may make sense to have Orphedomos build the image with the repo included, which would remove the need for this -->
-                        <alias>${docker.baseline.repo.id}/${dockerImageName}:${dockerImageVersion}</alias>
```
Comment (carter-cundiff) [inline extensions/extensions-docker/pom.xml:113]: I: Images now include the `${docker.baseline.repo.id}` by default so no need for the alias in fabric8.

File: build-parent/pom.xml [hunk 7]
```diff
@@ -541,6 +545,38 @@
...
+                    <artifactId>docker-maven-plugin</artifactId>
+                    <version>${version.fabric8.docker.maven.plugin}</version>
+                    <extensions>true</extensions> 
+                    <executions>
+                        <execution>
+                            <id>default-build</id>
+                            <configuration>
+                                <images>
+                                    <image>
+                                        <name>${docker.baseline.repo.id}/${dockerImageName}:${dockerImageVersion}</name>
```
Comment (carter-cundiff) [inline build-parent/pom.xml:559]: I: By including the `${docker.baseline.repo.id}` in our image name, `fabric8` knows to push the image to that docker repo. See [fabric8 docs](https://dmp.fabric8.io/#registry) for more details.

File: build-parent/pom.xml (hunk 1 above)
Comment (csun-cpointe) [reply build-parent/pom.xml:79]: Q: will this build both platforms for the downstream project?

File: extensions/extensions-docker/aissemble-model-training-api-containers/aissemble-model-training-api-sagemaker/pom.xml (hunk 2 above)
Comment (csun-cpointe) [inline extensions/extensions-docker/aissemble-model-training-api-containers/aissemble-model-training-api-sagemaker/pom.xml:46]: Q: are these the transitive dependencies that you are removing?

Comment (carter-cundiff) [reply extensions/extensions-docker/aissemble-model-training-api-containers/aissemble-model-training-api-sagemaker/pom.xml:46]: Some are transitive and some were just not used at all. The end result is we have on type `habushu` dependency which then contains the transitive ones.

**Review by csun-cpointe — APPROVED** [review]

**Review by d-ryan-ashcraft — APPROVED** [review]

**Pull request merged**